#
# Submitted-By: Robert Schwebel, 2006-10-17
# Committed-By: Robert Schwebel
#
# Error:
#
# FIXME: add real error description here
#
# Description:
#
# The autotools don't detect for architectures which don't have the
# ioprio_set syscall; if we are on such an architecture it just breaks
# during compilation.
#
# This is probably not the right fix, we might do a similar test as in
# schedutils/iosched.c, but during configure time and for the right
# architecture.
#
# State:
#
# unresolved as of 2.13-pre7
#

diff -urN util-linux-2.13-pre7/configure.ac util-linux-2.13-pre7-ptx/configure.ac
--- util-linux-2.13-pre7/configure.ac	2006-03-05 23:20:52.000000000 +0100
+++ util-linux-2.13-pre7-ptx/configure.ac	2006-10-16 22:52:13.000000000 +0200
@@ -161,13 +161,17 @@
 
 
 case "$host" in
+  arm*)    arm=true ;;
   i?86-*)  intel=true ;;
   m68*)    m68k=true ;;
+  s390*)   s390=true ;;
   sparc*)  sparc=true ;;
 esac
 
+AM_CONDITIONAL(ARM,   test x$arm   = xtrue)
 AM_CONDITIONAL(INTEL, test x$intel = xtrue)
 AM_CONDITIONAL(M68K,  test x$m68k  = xtrue)
+AM_CONDITIONAL(S390,  test x$s390  = xtrue)
 AM_CONDITIONAL(SPARC, test x$sparc = xtrue)
 
 
@@ -304,6 +308,13 @@
 )
 AM_CONDITIONAL(BUILD_SCHEDUTILS, test x$enable_schedutils = xyes)
 
+if test x$enable_schedutils = xyes; then
+  enable_ionice=no
+else
+  enable_ionice=yes
+fi
+AM_CONDITIONAL(BUILD_IONICE, test x$ionice = xyes)
+
 
 AC_ARG_ENABLE([wall],
   AC_HELP_STRING([--disable-wall], [do not build wall]),
diff -urN util-linux-2.13-pre7/schedutils/Makefile.am util-linux-2.13-pre7-ptx/schedutils/Makefile.am
--- util-linux-2.13-pre7/schedutils/Makefile.am	2005-08-14 13:33:22.000000000 +0200
+++ util-linux-2.13-pre7-ptx/schedutils/Makefile.am	2006-10-16 22:40:33.000000000 +0200
@@ -2,7 +2,13 @@
 
 if BUILD_SCHEDUTILS
 
-usrbinexec_PROGRAMS = chrt ionice taskset
-man_MANS = chrt.1 ionice.1 taskset.1
+usrbinexec_PROGRAMS = chrt taskset
+man_MANS = chrt.1 taskset.1
+
+if BUILD_IONICE
+usrbinexec_PROGRAMS += ionice
+man_MANS += ionice.1
+endif
+
 
 endif
